test/test_torch.py

```diff
@@ -341,14 +341,33 @@ def test_csub(self):
         res_csub.sub_(scalar)
         self.assertEqual(res_add, res_csub)
 
-    def test_neg(self):
-        a = torch.randn(100, 90)
-        zeros = torch.Tensor().resize_as_(a).zero_()
+    @staticmethod
+    def _test_neg(self, cast):
+        float_types = ['torch.DoubleTensor', 'torch.FloatTensor', 'torch.LongTensor']
+        int_types = ['torch.IntTensor', 'torch.ShortTensor']
+
+        for t in float_types + int_types:
```

Comment:
nit: not sure the separation into `float_types` and `int_types` are needed, wont `cast(torch.Tensor(100, 90).type(t).random_())` work for all of these types?